### PR TITLE
Automated tests and some PEP8

### LINKED ISSUE #59: Write automation test for last sprint

### PR DESCRIPTION
**Pull request by oudeismetis**
Fixes #59

### CODE REVIEW

File: franklin/builder/models.py [hunk 1]
```diff
@@ -232,11 +232,11 @@ def build(self):
...
         if not self.url:
             if self.name == self.site.DEFAULT_ENV:
-                self.url = "{0}.{1}".format(self.site.name,
+                self.url = "{0}.{1}".format(self.site.name.lower(),
                                             os.environ['BASE_URL'])
             else:
-                self.url = "{0}-{1}.{2}".format(self.site.name,
-                                                self.name,
+                self.url = "{0}-{1}.{2}".format(self.site.name.lower(),
+                                                self.name.lower(),
```
Comment (oudeismetis) [inline franklin/builder/models.py]: And a few bug fixes found while writing tests.

File: franklin/github/tests.py [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+        mock_file_data = mock.Mock()
+        mock_file_data.status_code = 200
+        mock_file_data.text = "hello: 'world'"
+        mock_get.side_effect = [mock_url, mock_file_data]
+
+        franklin_config = get_franklin_config(self.site, self.user)
+        if franklin_config and not hasattr(franklin_config, 'status_code'):
+            self.assertEqual(franklin_config.get('hello', None), 'world')
+        else:
+            self.fail('Received an HTTP Response instead of data from method')
```
Comment (oudeismetis) [inline franklin/github/tests.py]: This test ends up calling `make_rest_get_call` twice. So here we create 2 mock objects and add them to `side_effect`. The first call will return the first, 2nd returns 2nd, etc.

**Comment by JulianGindi** [comment]
:+1: always love new tests. Good stuff.

**Comment by camiriveron** [comment]
:+1:

**Pull request merged**